File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/domain/usecase/GetAllHelpcardsUseCase.java
package space.lopatkin.spb.helpboardgamecard.domain.usecase;

import androidx.lifecycle.LiveData;
import space.lopatkin.spb.helpboardgamecard.domain.repository.AppRepository;
import space.lopatkin.spb.helpboardgamecard.domain.model.Helpcard;

import java.util.List;

public class GetAllHelpcardsUseCase {

    private AppRepository repository;

    public GetAllHelpcardsUseCase(AppRepository repository) {
        this.repository = repository;
    }

    public LiveData<List<Helpcard>> execute() {
        return repository.getAllHelpcards();
    }

}

File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/domain/usecase/GetAllHelpcardsUseCase.kt
package space.lopatkin.spb.helpboardgamecard.domain.usecase

import androidx.lifecycle.LiveData
import space.lopatkin.spb.helpboardgamecard.domain.model.Helpcard
import space.lopatkin.spb.helpboardgamecard.domain.repository.AppRepository

class GetAllHelpcardsUseCase(private val repository: AppRepository) {

    fun execute(): LiveData<List<Helpcard>> {
        return repository.getAllHelpcards()
    }

}
